ibis/backends/impala/tests/test_sql.py

```diff
@@ -172,40 +172,47 @@ def test_nested_join_multiple_ctes():
     # projection from a filter was hiding an insidious bug, so we're disabling
     # that for now see issue #1295
     cond = joined3.movieid.isin(top_user_old_movie_ids.movieid)
-    result = joined3[cond]
+    filtered_expr = joined3[cond]
 
     expected = """\
-WITH t0 AS (
-  SELECT `userid`, `movieid`, `rating`,
-         CAST(`timestamp` AS timestamp) AS `datetime`
-  FROM ratings
-),
-t1 AS (
-  SELECT t0.*, t5.`title`
-  FROM t0
-    INNER JOIN movies t5
-      ON t0.`movieid` = t5.`movieid`
-)
-SELECT t2.*
+SELECT t0.*
 FROM (
-  SELECT t1.*
-  FROM t1
-  WHERE (t1.`userid` = 118205) AND
-        (extract(t1.`datetime`, 'year') > 2001)
-) t2
-WHERE t2.`movieid` IN (
+  SELECT t2.*
+  FROM (
+    SELECT t5.*, t4.`title`
+    FROM (
+      SELECT `userid`, `movieid`, `rating`,
+             CAST(`timestamp` AS timestamp) AS `datetime`
+      FROM ratings
+    ) t5
+      INNER JOIN movies t4
+        ON t5.`movieid` = t4.`movieid`
+  ) t2
+  WHERE (t2.`userid` = 118205) AND
+        (extract(t2.`datetime`, 'year') > 2001)
+) t0
+WHERE t0.`movieid` IN (
   SELECT `movieid`
   FROM (
-    SELECT t1.*
-    FROM t1
-    WHERE (t1.`userid` = 118205) AND
-          (extract(t1.`datetime`, 'year') > 2001) AND
-          (t1.`userid` = 118205) AND
-          (extract(t1.`datetime`, 'year') < 2009)
-  ) t4
+    SELECT t3.*
+    FROM (
+      SELECT t5.*, t4.`title`
+      FROM (
+        SELECT `userid`, `movieid`, `rating`,
+               CAST(`timestamp` AS timestamp) AS `datetime`
+        FROM ratings
+      ) t5
```

Comment:
It looks like this subquery is no longer extracted.

Reply:
Yes, don't know yet why.